zebra-state/src/service/finalized_state/disk_db.rs

```diff
@@ -533,16 +549,20 @@ impl DiskDb {
 
         let upper_bound = match range.end_bound().cloned() {
             Included(mut bound) => {
-                // Skip adding an upper bound if every byte is u8::MAX, or
-                // increment the last byte in the upper bound that is less than u8::MAX,
+                // Increment the last byte in the upper bound that is less than u8::MAX,
                 // and clear any bytes after it to increment the big-endian number this
                 // Vec represents to RocksDB.
-                let is_max_key = bound.iter_mut().rev().all(|v| {
+                let is_zero = bound.iter_mut().rev().all(|v| {
                     *v = v.wrapping_add(1);
                     v == &0
                 });
 
-                (!is_max_key).then_some(bound)
+                if is_zero {
+                    bound.push(0);
+                    *bound.get_mut(0).expect("should have at least 1 element") += 1;
```

Comment:
```suggestion
                    *bound.get_mut(0).expect("should have at least 1 element") = 1;
```

Reply:
Can't we just push 0x01 in the line before?

Reply:
> Can't we just push 0x01 in the line before?

I think that would put the 1 at the last element and that we need it in the first element.

Reply:
`bound.insert(0, 0x01)` achieves the same thing as these 2 lines, but it's much less verbose.